alg/gdaltransformer.cpp

```diff
@@ -467,51 +469,56 @@ GDALSuggestedWarpOutput2( GDALDatasetH hSrcDS,
         if( nSteps > 20 )
         {
             nSteps = 20;
+            nStepsPlusOne = 21;
             goto retry;
         }
         return CE_Failure;
     }
 
     padfY = padfX + nSampleMax;
-    padfZ = padfX + nSampleMax * 2;
+    padfZ = padfX + (nSampleMax<<1);
     padfYRevert = padfXRevert + nSampleMax;
-    padfZRevert = padfXRevert + nSampleMax * 2;
+    padfZRevert = padfXRevert + (nSampleMax<<1);
 
     // Take N_STEPS steps.
-    for( int iStep = 0; iStep <= nSteps; iStep++ )
+    double dfRatio = 0.0;
+    int iStep  = 0;
+    int iStep2 = 0;
```

Comment:
the scope of those variables should be kept inside the loop to minimize the number of things in the outer scope. compilers are smart enough to fold variables into registers